refiner-lib/src/refiner_inner.rs

```diff
@@ -626,6 +667,37 @@ fn fill_tx(
         .s(U256::zero())
 }
 
+/// Try to extract an address from an execution outcome status.
+fn extract_address(
+    status: &ExecutionStatusView,
+    tx_kind: &InnerTransactionKind,
+) -> Result<Option<Address>, RefinerError> {
+    if let ExecutionStatusView::SuccessValue(value_str) = status {
+        let value_vec = base64::decode(value_str).map_err(RefinerError::SuccessValueBase64Args)?;
+
+        let address_vec = match tx_kind {
+            InnerTransactionKind::DeployErc20 => value_vec,
+            InnerTransactionKind::Submit | InnerTransactionKind::Deploy => {
+                let submit_result = SubmitResult::try_from_slice(&value_vec)
+                    .map_err(|_| RefinerError::FailNearTx)?;
+                match submit_result.status {
+                    aurora_engine::parameters::TransactionStatus::Succeed(submit_result_vec) => {
+                        submit_result_vec
+                    }
+                    _ => return Err(RefinerError::FailNearTx),
```

Comment:
I think `Ok(None)` is better here. It is possible for the EVM to revert during contract creation, so the status will not always be `Succeed`.

Honestly, the unfortunate part is that still isn't even quite right. Since the contract address is independently knowable from the execution result, Ethereum tooling reports the address that would have been created if the execution didn't revert. Here is an [example](https://goerli.etherscan.io/tx/0xdeff004bc889227cd416f39014d352421e8a18ada67123ad295f80269fcc9a2f). So maybe what we are actually supposed to do is not extract the address from the outcome at all and instead use the Borealis Engine instance we have available to determine what the address should be instead.

Reply:
This is now the new approach, i.e using engine to get the address.